codalab/lib/bundle_cli.py

```diff
@@ -1455,43 +1455,50 @@ def copy_bundle(
             },
         )
 
-        # If bundle contents don't exist, finish after just copying metadata
+        # Fetch bundle contents from source client
         try:
             target_info = source_client.fetch_contents_info(BundleTarget(source_bundle_uuid, ''))
         except NotFoundError:
-            return
+            target_info = None
 
         # Collect information about how server should unpack
         filename = nested_dict_get(source_info, 'metadata', 'name')
-        if target_info['type'] == 'directory':
+        # Zip bundle directory if there is any
+        if target_info and target_info['type'] == 'directory':
             filename += '.tar.gz'
             unpack = True
         else:
             unpack = False
 
-        # Bundles stuck in non-final states such as 'running' should not keep
-        # that state at the destination server, and should instead just fallback
-        # to 'failed'
-        if source_info['state'] == State.READY:
-            source_state = State.READY
+        try:
```

Comment:
Can we just check if `target_info is None` to avoid making a failing request?

Reply:
Updated.